aea/protocols/base.py

```diff
@@ -225,15 +226,18 @@ def decode(self, obj: bytes) -> Message:
         return Message(json_msg)
 
 
-class Protocol(ABC):
+class Protocol(Component):
     """
     This class implements a specifications for a protocol.
 
     It includes a serializer to encode/decode a message.
     """
 
     def __init__(
-        self, protocol_id: ProtocolId, serializer: Serializer, config: ProtocolConfig
+        self,
+        protocol_id: Optional[ProtocolId] = None,
```

Comment:
why all optional?

Reply:
ok let's break backwards compatibility here too and just initialize as `None` below (with type declaration)